#!/usr/bin/env python
PACKAGE = "olelidar"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


# device
gen.add("device_ip", str_t, 0, "IP addr to receive packet from.", "192.168.1.100")
gen.add("device_port", int_t, 0, "UDP port to receive packet from.", 2368)
gen.add("freq", double_t, 0, "rotate degree per second", 15.0)
gen.add("route", int_t, 0, "rotate degree per second", 2000)
gen.add("step", double_t, 0, "rotate degree per second", 0.225)
# laserscan
'''
std_msgs/Header header
      uint32 seq
      time stamp
      string frame_id
float32 angle_min
float32 angle_max
float32 angle_increment
float32 time_increment
float32 scan_time
float32 range_min
float32 range_max
float32[] ranges
float32[] intensities
'''

gen.add("frame_id", str_t, 0, "olelidar")
gen.add("angle_min", double_t, -180, "angle_min",0,0,-180)
gen.add("angle_max", double_t, 180, "angle_max",360,0,180)
gen.add("range_min", double_t, 0, "min range", 0.1, 0.1, 20)
gen.add("range_max", double_t, 0, "max range", 50.0, 0.1, 50)

exit(gen.generate(PACKAGE, PACKAGE, "OleiPuck"))
